### DISCUSSION TITLE
Ubuntu and modules

### DISCUSSION THREAD

**Discussion by zekdemir** [minios-linux/minios-live#19] [Q&A] [answered]
Hi, dear developers! First, I want to thank you for providing MiniOS. It's great and I use it my as my daily system. I am not an experienced linux user so I have 2 questions for you regarding MiniOS:

1- I used the MiniOS script to build a live Ubuntu on a variety of Debian and Ubuntu spins but it always failed giving some "ubuntu mos" related errors. What could be wrong? Should I be using original Ubuntu? I think the error was related to not being able to connect to ubuntu mos server or something like that.

2- Can I build a module from unofficial repos? I tried to build a TDE (Trinity Desktop Environment) module but it failed saying that it was "Unable to locate package tde-trinity". I was able to add the unofficial TDE repo and install the desktop in the usual way but I want to build a module of it instead similar to the XFCE module you provide. The script seems to be not using the installed repos as usual "apt install tde-trinity" command works without problems.

Thanks again!..

**Comment by crim50n** [comment]
1. Only Jammy-based MiniOS Standard with default settings can be guaranteed to be built on Ubuntu, other settings may not work.
2. Do you want to build modules using build scripts or using utilities in a running MiniOS?

**Reply by zekdemir** [reply]
I tried to build the module using the build scripts on a terminal screen in a running MiniOS. Is it possible in other ways too? What utilities are there?

**Reply by crim50n** [reply, marked as answer]
To build the modules, use the instructions at the following link: https://github.com/minios-linux/minios-live/wiki/Creating-modules
